aptos-move/framework/aptos-framework/sources/jwks.move

```diff
@@ -459,10 +462,42 @@ module aptos_framework::jwks {
     public fun upsert_into_observed_jwks(fx: &signer, provider_jwks_vec: vector<ProviderJWKs>) acquires ObservedJWKs, PatchedJWKs, Patches {
         system_addresses::assert_aptos_framework(fx);
         let observed_jwks = borrow_global_mut<ObservedJWKs>(@aptos_framework);
-        vector::for_each(provider_jwks_vec, |obj| {
-            let provider_jwks: ProviderJWKs = obj;
-            upsert_provider_jwks(&mut observed_jwks.jwks, provider_jwks);
-        });
+
+        if (features::is_jwk_consensus_per_key_mode_enabled()) {
+            vector::for_each(provider_jwks_vec, |proposed_provider_jwks|{
+                let maybe_cur_issuer_jwks = remove_issuer(&mut observed_jwks.jwks, proposed_provider_jwks.issuer);
+                let cur_issuer_jwks = if (option::is_some(&maybe_cur_issuer_jwks)) {
+                    option::extract(&mut maybe_cur_issuer_jwks)
+                } else {
+                    ProviderJWKs {
+                        issuer: proposed_provider_jwks.issuer,
+                        version: 0,
+                        jwks: vector[],
+                    }
+                };
+                assert!(cur_issuer_jwks.version + 1 == proposed_provider_jwks.version, error::invalid_argument(EUNEXPECTED_VERSION));
+                vector::for_each(proposed_provider_jwks.jwks, |jwk|{
```

Comment:
is the vector length always 1 or?

Reply:
yes, currently the upstream ensures the length is 1. No need to assert here i think.